lib/src/mock_field_value_platform.dart

```diff
@@ -1,22 +1,31 @@
 import 'package:cloud_firestore_platform_interface/cloud_firestore_platform_interface.dart';
+import 'package:fake_cloud_firestore/src/fake_server_time_provider.dart';
 import 'package:fake_cloud_firestore/src/util.dart';
 import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 
 abstract class FakeFieldValue {
   const FakeFieldValue();
 
   static const delete = FieldValueDelete();
-  static const serverTimestamp = FieldValueServerTimestamp();
+  static final serverTimestamp = FieldValueServerTimestamp();
 
   void updateDocument(Map<String, dynamic> document, String key);
 }
 
 class FieldValueServerTimestamp extends FakeFieldValue {
-  const FieldValueServerTimestamp();
+  FakeServerTimeProvider? _fakeServerTimeProvider;
+
+  Timestamp get now => _fakeServerTimeProvider?.now ?? Timestamp.now();
+
+  void setTimeProvider(FakeServerTimeProvider? fakeServerTimeProvider) {
+    _fakeServerTimeProvider = fakeServerTimeProvider;
+  }
+
+  FieldValueServerTimestamp();
 
   @override
   void updateDocument(Map<String, dynamic> document, String key) {
-    document[key] = Timestamp.now();
+    document[key] = now;
```

Comment:
With the Clock always set up, you can rewrite this as

```dart
document[key] = clock.now()
```

https://pub.dev/documentation/clock/latest/clock/Clock/now.html